x/bounty/keeper/finding.go

```diff
@@ -0,0 +1,100 @@
+package keeper
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/shentufoundation/shentu/v2/x/bounty/types"
+)
+
+func (k Keeper) GetFinding(ctx sdk.Context, id uint64) (types.Finding, bool) {
+	store := ctx.KVStore(k.storeKey)
+
+	pBz := store.Get(types.GetFindingKey(id))
+	if pBz == nil {
+		return types.Finding{}, false
+	}
+
+	var finding types.Finding
+	k.cdc.MustUnmarshal(pBz, &finding)
+	return finding, true
+}
+
+func (k Keeper) SetFinding(ctx sdk.Context, finding types.Finding) {
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshal(&finding)
+	store.Set(types.GetFindingKey(finding.FindingId), bz)
+}
+
+func (k Keeper) GetNextFindingID(ctx sdk.Context) uint64 {
+	store := ctx.KVStore(k.storeKey)
+	Bz := store.Get(types.GetNextFindingIDKey())
+	return binary.LittleEndian.Uint64(Bz)
+}
+
+func (k Keeper) SetNextFindingID(ctx sdk.Context, id uint64) {
```

Comment:
Discussion about storage:
The findingID storage corresponding to programID
Is there a need for a del operation (e.g. withdrawing the finding)? , or use key:valye instead of []uint64? :
1. key:(prefix+pId+fId), value:(fId)
2. When we store, we don't need to get all the FId and then append
3. When obtaining all FId corresponding to PId, over all the keys with a certain prefix+pid.

Reply:
It's not necessary to read all the FIDs under the PID every time when save the FID.

Reply:
We need to delete the finding when withdrawing it.

Reply:
We can find the key info (prefix+pld+fld) for delete op.

Reply:
This will take up more storage space.